test/pop_manager_test.erl

```diff
@@ -115,37 +121,80 @@ extract_new_block_messages() ->
 	    []
     end.
 
+extract_send_full_blocks() ->
+    receive
+	{net, send_full_blocks, {old, [Block]}} -> 
+	    [Block | extract_send_full_blocks()];
+	_Any ->
+	    fail
+    after 0 -> 
+	    []
+    end.
+    
+
 no_more_messages() ->
     receive
 	_Any ->
 	    false
     after 0 -> 
 	    true
     end.
+
+make_transaction(PrivateKey, PublicKey, Nonce, ChainId) ->
+    T0 = #{
+	   game_data => undefined,
+	   nonce => Nonce,
+	   player_id => PublicKey,
+	   consensus_data => #{
+			       signature => undefined,
+			       chain_id => ChainId
+			      }
+	  },
+    Hash = my_crypto:hash(my_serializer:serialize_object(T0)),
+    Signature = my_crypto:sign(Hash, PrivateKey),
+    CD = maps:get(consensus_data, T0),
+    T1 = T0#{
```

Comment:
should there be a helper function to hash + sign a block?

Reply:
eh, probably at some point